### ISSUE TITLE
Problem with tools.py

### ISSUE THREAD

**Issue by pclinuxer** [bit-team/backintime#1169]
Hello:

I run a Devuan Linux Beowulf box on a backported kernel ...

```
groucho@devuan:~$ uname -a
Linux devuan 5.10.0-0.bpo.3-amd64 #1 SMP Debian 5.10.13-1~bpo10+1 (2021-02-11) x86_64 GNU/Linux
groucho@devuan:~$ 

```
... and use backintime  v. 1.1.24, having done so without any issues for the longest while and with many other distributions before Devuan.

Today I received an error message in my mailbox which I have never seen before:

```
--- snip ---
Unhandled exception in thread started by <function __log_keyring_warning at 0x7fdf90288488>
Traceback (most recent call last):
File "/usr/share/backintime/common/tools.py", line 1458, in __log_keyring_warning
TypeError: 'NoneType' object is not callable
```

This is the part of the file `/usr/share/backintime/common/tools.py` the message refers to, with the line numbers as rendered by `jed`:

```
1456
1457 def __log_keyring_warning():
1458     from time import sleep
1459     sleep(0.1)
1460     logger.warning('import keyring failed')
1461
1462 if keyring is None and keyring_warn:
1463     #delay warning to give logger some time to import
1464     import _thread
1465     _thread.start_new_thread(__log_keyring_warning, ())
1466     # logger.warning('import keyring failed')
1467                                                         
```
The notification sent by the `Cron Daemon` has the timestamp Mon, 21 Jun 2021 11:00:02 -0300.
The last `backintime` log entry was for a snapshot with timestamp Mon Jun 21 07:15:01 2021 local time and does not show any errors.

The crontab system entry script has not been edited or modified. 

```
# Back In Time system entry, this will be edited by the gui:
60 */15 * * * * /usr/bin/nice -n 19 /usr/bin/ionice -c2 -n7 /usr/bin/backintime backup-job >/dev/null

```
Installed Python version is 2.7.16 (default, Oct 10 2019, 22:02:15).

**EDIT:**
Looking around, I found a report from 10/2017, ie: four years ago.
It is (for all intent and purposes) identical to what I have reported here.

See: [(https://github.com/bit-team/backintime/issues/820)](url)

The OP cites the same thing: happens very rarely.
In my case, I had never seen it before.
He also posted the output from `--debug` but there was never any follow up to that.

This was happening as he ran backintime 1.1.20 under Linux 4.4.87-18.29-default x86_64. (Open Suse).
In my case, it happened running backintime 1.1.24 under Linux 5.10.0-0.bpo.3-amd64 (Devuan Beowulf). 

Any idea as to what this is about?

Thanks in advance,

 PCL

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
Older (duplicated) report: #820

**Comment by emtiu** [comment]
Thank you for the report, @pclinuxer, I'm closing this as a duplicate of #820.

**Comment by pclinuxer** [comment]
Hello:
> Thank you for the report, @pclinuxer ...
You're welcome.
> I'm closing this as a duplicate of #820.
Yes, I did mention that I had found #820 (from 10/2017) when I posted a year ago.

#1169 being closed for being a duplicate of #820, makes total sense.

But I see that #820 has *also* been closed but there's no mention of a fix for it.
BTW; I am *still* getting the same unhandled exception every so often.
ie: the same problem as before.

code
~$ uname -a
Linux devuan 5.10.0-0.bpo.15-amd64 #1 SMP Debian 5.10.120-1~bpo10+1 (2022-06-13) x86_64 GNU/Linux
 ~$ 
/code

code
~$ python -V
Python 2.7.16
~$ 
/code

Please advise.

Thanks in advance.

Best,

PCL
